### PR TITLE
Add workflowGraphs view and fix unit-test

### PR DESCRIPTION
**Pull request by srinia6**
Add view to workflowGraphs and fix unit-test.

@brianparry @BillyAbildgaard @keedya @dalebremner @RackHD/corecommitters

### CODE REVIEW

**Comment by srinia6** [comment]
Above error is due to views not available in the first commit. Will try to re-run

**Comment by changev** [comment]
Jenkins: test this please

**Comment by changev** [comment]
Jenkins: test this please

**Comment by changev** [comment]
test this please

File: data/views/workflowGraphs.2.0.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <% }%>
+    "tasks":[
+    <% tasks.forEach(function (value, i, arr){ %>
+    {
+    "label": "<%=value.label%>",
+    <% if ( value.waitOn) { %>
+    "waitOn": <%- JSON.stringify(value.waitOn) %>,
+    <% } %>
+
+    <% if ( value.taskDefinition ) { %>
```
Comment (yyscamper) [inline data/views/workflowGraphs.2.0.json]: Please add some unit-test case for inline task definition

**Comment by yyscamper** [comment]
@srinia6 : Last Friday, I opened a PR to add unit-test for template files (see https://github.com/RackHD/on-http/pull/427), could you please remember to take an action for the new view after my PR is merged?

File: data/views/workflowGraphs.2.0.json (hunk 1 above)
Comment (srinia6) [inline data/views/workflowGraphs.2.0.json]: @yyscamper We don't have unit-test for views.

**Comment by srinia6** [comment]
@yyscamper Not sure what action you are referring to w.r.t after your PR is merged - https://github.com/RackHD/on-http/pull/427 . Are you talking about unit-test?

**Comment by srinia6** [comment]
@changev: Is there a reason to re-test this PR? ALL tests were passing last when i left off.

**Comment by srinia6** [comment]
test this please

**Comment by yyscamper** [comment]
@srinia6 : yes, I mean the unit test

**Comment by brianparry** [comment]
Looks good @srinia6 :+1:

**Comment by keedya** [comment]
:+1:

File: data/views/workflowGraphs.2.0.json (hunk 1 above)
Comment (yyscamper) [inline data/views/workflowGraphs.2.0.json]: That's why I opened the PR for template unit-testing: https://github.com/RackHD/on-http/pull/427

**Comment by yyscamper** [comment]
@srinia6 @brianparry : I see you have some concern on the unit-testing for view templates. If I don't persuade you at present, please at least add an unit-testing in API to test the GET /workflows/graphs can also return correct rendered data for inline task definition.

**Comment by yyscamper** [comment]
:+1: after adding test cases for inline task definition.

File: spec/lib/api/2.0/workflowGraphs-spec.js [hunk 2]
```diff
@@ -37,29 +37,118 @@ describe('Http.Api.Workflows.2.0', function () {
...
+
+                    expect(res.body[0])
+                        .to.have.property('friendlyName').that.equals('Dummy Pollers');
+
+                    expect(res.body[0])
+                        .to.have.property('injectableName').that.equals('Dummy.Poller.Create');
+                });
+        });
+
+        it('should retrieve the workflow Graphs with inline task', function () {
```
Comment (yyscamper) [inline spec/lib/api/2.0/workflowGraphs-spec.js]: Please fix all indent problems within this test cases.

Comment (srinia6) [inline spec/lib/api/2.0/workflowGraphs-spec.js]: These indent were existing in this file.

**Comment by srinia6** [comment]
Consolidating 4 commits to 1.

File: spec/lib/api/2.0/workflowGraphs-spec.js (hunk 2 above)
Comment (srinia6) [inline spec/lib/api/2.0/workflowGraphs-spec.js]: DONE

File: spec/lib/api/2.0/workflowGraphs-spec.js [hunk 3]
```diff
@@ -37,29 +37,118 @@ describe('Http.Api.Workflows.2.0', function () {
...
+            var graph =
+            {
+                "friendlyName": "Dummy Pollers",
+                "injectableName": "Dummy.Poller.Create",
+                "tasks": [
+                    {
+                        "label": "boot-graph",
+			"taskDefinition": {
+			"friendlyName": "Boot Graph",
+			"injectableName": "Task.Graph.Run.Boot",
```
Comment (yyscamper) [inline spec/lib/api/2.0/workflowGraphs-spec.js]: still indent problem

File: spec/lib/api/2.0/workflowGraphs-spec.js [hunk 4]
```diff
@@ -37,29 +37,118 @@ describe('Http.Api.Workflows.2.0', function () {
...
+                ]
+            };
+            var task = {
+                "friendlyName": "Boot Graph",
+                "injectableName": "Task.Graph.Run.Boot",
+                "implementsTask": "Task.Base.Graph.Run",
+                "options": {
+                    "graphName": "Graph.BootstrapUbuntu",
+                    "defaults": {
+                    "graphOptions": {}
```
Comment (yyscamper) [inline spec/lib/api/2.0/workflowGraphs-spec.js]: still indent problem

Comment (srinia6) [inline spec/lib/api/2.0/workflowGraphs-spec.js]: I fixed a trivial curl braces indent just now.

**Comment by benbp** [comment]
:+1: I think it's fine to do unit tests as a follow-on rather than block this, since it's not an implemented framework yet and the design is still under some disagreement from my understanding.

**Pull request merged**